src/auth/azure.rs

```diff
@@ -0,0 +1,84 @@
+use crate::auth::store_access_token;
+use crate::IdentityProvider;
+use azure_identity::{authorization_code_flow, development};
+use oauth2::{AccessToken, ClientId, RefreshToken, TokenResponse};
+use std::env;
+use std::error::Error;
+use url::Url;
+
+pub async fn azure_authorize(
+    scope: String,
+    refresh_token: Option<RefreshToken>,
+) -> Result<AccessToken, Box<dyn Error>> {
+    let client_id = ClientId::new(
+        env::var("OAUTH_CLIENT_ID").expect("Missing CLIENT_ID environment variable."),
+    );
+    let tenant_id = env::var("AZURE_TENANT_ID").expect("Missing TENANT_ID environment variable.");
+    if let Some(refresh_token) = refresh_token {
+        let token = azure_refresh_authorization(&refresh_token, &client_id, &tenant_id).await;
+        match token {
+            Ok(token) => return Ok(token),
+            Err(e) => eprintln!("{}", e),
+        }
+    }
+    azure_browser_authorization(scope, client_id, tenant_id).await
+}
+
+async fn azure_browser_authorization(
+    scope: String,
+    client_id: ClientId,
+    tenant_id: String,
+) -> Result<AccessToken, Box<dyn Error>> {
+    let scope = format!("{} offline_access", scope);
+
+    let c = authorization_code_flow::start(
+        client_id,
+        None,
+        &tenant_id,
+        Url::parse("http://localhost:47471")?,
+        &*scope,
+    );
+    println!("\nOpen this URL in a browser:\n{}", c.authorize_url);
+
+    // Using a naive, blocking redirect server is fine for our case, as it should block anyway
+    let code = development::naive_redirect_server(&c, 47471)?;
```

Comment:
Hi @KovacicAi, can you provide a link to those updates?

Reply:
Hi @mattphelps-8451! Sure thing.
I was thinking along the lines of the Azure CLI Credential. 
I wasn't able to get that to work at the time, but I feel that we should at least provide the option of using it as it could make the use of the CLI even easier for people who are logged into the Azure CLI.
My main idea is that we default to the CLI credential, and use Auth Code flow as a fallback, and we could possibly add a flag to force either one. I think that could be made an issue though.

I know I'm flip-flopping, but I think this auth code flow is fine as-is right now, though it will need some minor modification as we update to azure_identity 0.6.0.

Reply:
@KovacicAi I think this is a solid reminder to not let perfect be the enemy of the good! This PR cleans up a lot of the code. We don't need to tackle it all at once. Let's keep these auth code flow ideas in the back of our minds for the next enhancement.